src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -161,6 +110,34 @@ public FileResource(FileService fileService, ResourceLoader resourceLoader, Lect
         return responseEntityForFilePath(FilePathService.getTempFilepath() + filename);
     }
 
+    /**
+     * POST /markdownFileUpload : Upload a new file for markdown.
+     *
+     * @param file The file to save
+     * @param keepFileName specifies if original file name should be kept
+     * @return The path of the file
+     * @throws URISyntaxException if response path can't be converted into URI
+     */
+    @PostMapping("/markdownFileUpload")
+    @PreAuthorize("hasAnyRole('ADMIN', 'INSTRUCTOR', 'TA', 'USER')")
```

Comment:
We should think about the permission concept here. For now I suggest that we only allow this to ADMIN, INSTRUCTOR and TA roles.

Reply:
This also means the image upload should be deactivate for the questions and answer component for now

Reply:
I deactivated it for questions and answers and removed the user role